### PR TITLE
fix(components): [date-picker] Return to normal when reopened

### LINKED ISSUE #11612: [Component] [date-picker] After the time picker selects the time, open the selection again and hover, the time picker and the value display are inconsistent
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.45`
- Element Plus Version: `2.2.27`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36 Edg/108.0.1462.54`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1kYXRlLXBpY2tlclwiPlxuICAgIDxkaXYgY2xhc3M9XCJibG9ja1wiPlxuICAgICAgPHA+Q29tcG9uZW50IHZhbHVl77yae3sgdmFsdWUgfX08L3A+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICAgICAgdHlwZT1cImRhdGVyYW5nZVwiXG4gICAgICAgIHZhbHVlLWZvcm1hdD1cIllZWVktTU0tRERcIlxuICAgICAgICBzdGFydC1wbGFjZWhvbGRlcj1cIlN0YXJ0IGRhdGVcIlxuICAgICAgICBlbmQtcGxhY2Vob2xkZXI9XCJFbmQgZGF0ZVwiXG4gICAgICAvPlxuICAgIDwvZGl2PlxuICA8L2Rpdj5cbiAgPG9sPlxuXHRcdDxsaT5TZWxlY3QgdGhlIHN0YXJ0IGRhdGUgYW5kIGVuZCBkYXRlIChmb3IgZXhhbXBsZSwgMjAyMy0wMi0xMCwgMjAyMy0wMi0xNSk7PC9saT5cbiAgICA8bGk+T3BlbiB0aGUgZGF0ZSBwaWNrZXIgYWdhaW4sIG9ubHkgY2xpY2sgb25jZSAoaWYgaXQgaXMgMjAyMy0wMi0yMCksIGFuZCB0aGVuIGhvdmVyIHRoZSBtb3VzZSBvdmVyIGFub3RoZXIgZGF0ZSAoc3VjaCBhcyAyMDIzLTAyLTI1LCBidXQgZG8gbm90IGNsaWNrISEpLCBhbmQgZmluYWxseSBjbGljayBvdXRzaWRlIHRoZSBkYXRlIHBpY2tlciB0byBjbG9zZSBpdC47PC9saT5cblx0XHQ8bGk+UmVvcGVuIHRoZSBkYXRlIHBpY2tlciwgd2UgY2FuIGZpbmQgdGhhdCB0aGUgc2VsZWN0ZWQgZGF0ZSBvbiB0aGUgZGF0ZSBwaWNrZXIgaXMgZGlmZmVyZW50IGZyb20gb3VyIGNvbXBvbmVudCB2YWx1ZS4gT3VyIGNvbXBvbmVudCB2YWx1ZSBpcyBb4oCcMjAyMy0wMi0xMOKAnSwg4oCcMjAyMy0wMi0xNeKAnV0sIGJ1dCB0aGUgaGlnaGxpZ2h0ZWQgdGltZSBpbiB0aGUgdGltZSBwaWNrZXIgaXMgMjAyMy0wMi0yMCBhbmQgMjAyMy0wMi0yNSE8L2xpPlxuICA8L29sPlxuICA8cD5cbiAgICBFeHBlY3RlZCByZXN1bHQ6IFRoZSBkYXRlIHBpY2tlciBhbHdheXMgc3RheXMgY29uc2lzdGVudCB3aXRoIHRoZSBjb21wb25lbnQgdmFsdWUuXG4gIDwvcD5cbiAgXG4gIOS4reaWh++8mlxuICA8b2w+XG4gICAgPGxpPumAieS4reW8gOWni+aXpeacn+WSjOe7k+adn+aXpeacn++8iOS+i+WmgjIwMjMtMDItMTDvvIwyMDIzLTAyLTE177yJ77ybPC9saT5cbiAgICA8bGk+5YaN5qyh5omT5byA5pel5pyf6YCJ5oup5Zmo77yM5Y+q54K55Ye75LiA5qyh77yI5YGH5aaC5pivMjAyMy0wMi0yMO+8ie+8jOeEtuWQjum8oOagh+aCrOenu+WIsOWPpuS4gOS4quaXpeacn++8iOavlOWmgjIwMjMtMDItMjXvvIzkvYbmmK/kuI3opoHngrnlh7vvvIHvvIHvvInvvIzmnIDlkI7ngrnlh7vml6XmnJ/pgInmi6nlmajlpJbpg6jkvY3nva7ku6XlhbPpl63lroPjgILvvJs8L2xpPlxuICAgIDxsaT7ph43mlrDmiZPlvIDml6XmnJ/pgInmi6nlmajvvIzmiJHku6zlj6/ku6Xlj5HnjrDml6XmnJ/pgInmi6nlmajkuIrpnaLpgInkuK3nmoTml6XmnJ/lkozmiJHku6znmoTnu4Tku7blgLzkuI3kuIDmoLfjgILmiJHku6znmoTnu4Tku7blgLzmmK9bXCIyMDIzLTAyLTEwXCIsIFwiMjAyMy0wMi0xNVwiIF3vvIzkvYbmmK/lnKjml7bpl7TpgInmi6nlmajlhoXpq5jkuq7nmoTml7bpl7TmmK8yMDIzLTAyLTIw5Lul5Y+KMjAyMy0wMi0yNe+8gTwvbGk+XG4gIDwvb2w+XG48cD5cbiAgICDmnJ/lvoXnu5PmnpzvvJrml6XmnJ/pgInmi6nlmajkuIrmsLjov5zlkoznu4Tku7blgLzkv53mjIHkuIDoh7TjgIJcbiAgPC9wPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoJycpXG5jb25zdCBkZWZhdWx0VGltZSA9IHJlZjxbRGF0ZSwgRGF0ZV0+KFtdKVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZS1waWNrZXIge1xuICBkaXNwbGF5OiBmbGV4O1xuICB3aWR0aDogMTAwJTtcbiAgcGFkZGluZzogMDtcbiAgZmxleC13cmFwOiB3cmFwO1xufVxuLmRlbW8tZGF0ZS1waWNrZXIgLmJsb2NrIHtcbiAgcGFkZGluZzogMzBweCAwO1xuICB0ZXh0LWFsaWduOiBjZW50ZXI7XG4gIGJvcmRlci1yaWdodDogc29saWQgMXB4IHZhcigtLWVsLWJvcmRlci1jb2xvcik7XG4gIGZsZXg6IDE7XG59XG4uZGVtby1kYXRlLXBpY2tlciAuYmxvY2s6bGFzdC1jaGlsZCB7XG4gIGJvcmRlci1yaWdodDogbm9uZTtcbn1cbjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

1. Select the start date and end date (for example, 2023-02-10, 2023-02-15);
2. Open the date picker again, only click once (if it is 2023-02-20), and then hover the mouse over another date (such as 2023-02-25, but do not click!!), and finally click outside the date picker to close it.;
3. Reopen the date picker, we can find that the selected date on the date picker is different from our component value. Our component value is [“2023-02-10”, “2023-02-15”], but the highlighted time in the time picker is 2023-02-20 and 2023-02-25!

### What is Expected?

The date picker always stays consistent with the component value.

### What is actually happening?

The actual value and the date picker selected state are inconsistent

### Additional comments

_(empty)_

image

### PR DESCRIPTION
**Pull request by lyric-zemin**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix: https://github.com/element-plus/element-plus/issues/11612

### CODE REVIEW

**Comment by ubbcou** [comment]
Hi, I see you fixed the problem, but why shut down PR?

**Comment by ubbcou** [comment]
@lyric-zemin @btea @tolking 

Is there anything we can do?

**Comment by btea** [comment]
I also don't know why he closed this pr. It doesn't look like he responded to your comment either, I'll reopen this pr and try to review it.

**Review by btea — APPROVED** [review]

**Comment by lyric-zemin** [comment]
@ubbcou Sorry it took so long to reply, because I feel the repair is not elegant enough

**Comment by ubbcou** [comment]
Just do your best.

File: packages/components/date-picker/src/date-picker-com/panel-date-range.vue [hunk 1]
```diff
@@ -321,6 +322,17 @@ const {
...
+const attrs = useAttrs()
+watch(
+  () => attrs.visible,
+  (visible) => {
+    if (visible) {
+      onReset(props.parsedValue)
+      onSelect(false)
+    }
+  }
+)
```
[the remark this answers is not in the document]
Comment (lyric-zemin) [reply packages/components/date-picker/src/date-picker-com/panel-date-range.vue]: Thanks, submitted

**Review by tolking — APPROVED** [review]

**Pull request merged**